consensus/src/experimental/commit_phase.rs

```diff
@@ -270,54 +203,79 @@ impl CommitPhase {
             ledger_info.ledger_info().consensus_data_hash(),
         );
 
-        let sig = self
+        let signature = self
             .safety_rules
             .lock()
-            .sign_commit_vote(ledger_info.clone(), new_ledger_info.clone())?;
+            .sign_commit_vote(ledger_info, new_ledger_info.clone())?;
+
         // if fails, it needs to resend, otherwise the liveness might compromise.
-        self.commit_msg_sender
-            .send(CommitPhaseMessage::CommitVote(
-                self.author,
-                new_ledger_info.clone(),
-                sig,
-            ))
-            .await?;
-        // note that this message will also reach the node itself
 
-        self.local_queue
-            .push_back(PendingBlocks::new(vecblock, ledger_info));
+        let mut commit_sender = self.network_sender.clone();
+        let author = self.author;
+        let msg = ConsensusMsg::CommitVoteMsg(Box::new(CommitVote::new_with_signature(
+            author,
+            new_ledger_info.clone(),
+            signature.clone(),
+        )));
+
+        let mut new_ledger_info_with_sig = LedgerInfoWithSignatures::new(
+            new_ledger_info,
+            BTreeMap::<AccountAddress, Ed25519Signature>::new(),
+        );
+        new_ledger_info_with_sig.add_signature(self.author, signature);
+
+        self.blocks = Some((vecblock, new_ledger_info_with_sig));
+
+        // asynchronously broadcast the message.
+        // note that this message will also reach the node itself
+        tokio::spawn(async move {
```

Comment:
network sender is asynchronous anyway, we don't need to spawn a task

Reply:
resolved